airflow/www/views.py

```diff
@@ -527,6 +527,21 @@ def get_task_stats_from_query(qry):
     return data
 
 
+def redirect_or_json(origin, return_as_json, msg, status=""):
```

Comment:
I think we can remove `return_as_json` entirely and check `request.headers['Accept'] == 'application/json'` in this function directly

Reply:
Yeah, I didn't like that variable name. Updated.

Reply:
Actually you don’t even need to pass in `request`; the variable is magical in Flask and will automatically refer to the correct thing when it’s called in a view function.

Reply:
Hmm. That seemed to cause some testing errors with `KeyError: 'HTTP_ACCEPT'`. Would you have any idea on how to address that?